### PR TITLE
[XPU] Update XPU C Shim Header

### LINKED ISSUE #141268: xpu: build fails for XPU backend due to outdated aoti_torch/generated/c_shim_xpu.h
Building at the a84779040049377aec7b62e37becb7327950541e with XPU backend support fails due to outdated [torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h](https://github.com/pytorch/pytorch/blob/a84779040049377aec7b62e37becb7327950541e/torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h). See below log.

Issue was introduced by the following 2 commits both updating `torchgen/aoti/fallback_ops.py`:
* 34b2165bdb5 Insert aten.add into fallback_ops, and fix Tensor -> Scalar conversion in ir.FallbackKernel (#140624)
* 34e420519df [Reland] dont decompose baddbmm (#141045)

Pending update of torch-xpu-ops commit pin (#141201) did notice this issue.

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/dvrogozh/git/pytorch/pytorch/torchgen/gen.py", line 3042, in <module>
    main()
  File "/home/dvrogozh/git/pytorch/pytorch/torchgen/gen.py", line 2979, in main
    gen_source_files(
  File "/home/dvrogozh/git/pytorch/pytorch/torchgen/gen.py", line 2403, in gen_source_files
    assert old_header == new_header, """
AssertionError:

WARNING: The generated AOTInductor C shim header files have unexpectedly changed. This
indicates an AOTInductor fallback operator ABI backward compatibility breakage!!!
Only in a limited number of situations, this is allowed:

1. You added a fallback op to the inductor_fallback_ops list in torchgen/aoti/fallback_ops.py.
If that's the case, run `python torchgen/gen.py --update-aoti-c-shim` to update the existing
C shim header files.

2. You added a new default argument to an existing fallback op. This is clearly a BC breaking
change in the AOTInductor land. In this case, you need to keep a manual copy of that existing
fallback op in a file, e.g. torch/csrc/inductor/aoti_torch/c/shim.h, bump up the version
number of that fallback op in the newly generated C shim files, and update the cpp wrapper
codegen to generate the correct cpp call for this op. Contact AOTInductor team for assistance.


CMake Error at cmake/Codegen.cmake:218 (message):
  Failed to get generated_sources list
Call Stack (most recent call first):
  caffe2/CMakeLists.txt:2 (include)
```

CC: @gujinghui @EikanWang @fengyuan14 @guangyey @jgong5 @xytintel 

cc @malfet @seemethere @gujinghui @EikanWang @fengyuan14 @guangyey

### PR DESCRIPTION
**Pull request by ratnampa**
Fixes https://github.com/pytorch/pytorch/issues/141268

Caused by these commits: https://github.com/pytorch/pytorch/commit/34b2165bdb5d84a8ffe97f5348394c8ef0b5c6d8 and https://github.com/pytorch/pytorch/commit/34e420519df9c711ac997888470022cd7c4a90f6

The windows XPU builds are failing: https://github.com/pytorch/pytorch/actions/runs/11922274722/job/33228175750 
due to recent PR merge with changes in fallback ops: https://github.com/pytorch/pytorch/commit/34e420519df9c711ac997888470022cd7c4a90f6

This PR updates the XPU C Shim header file to overcome these build failures.

### CODE REVIEW

**Review by etaf — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

**Review by jansel — CHANGES_REQUESTED** [review]
Failing tests?

**Comment by etaf** [comment]
> Failing tests?

Hi, @jansel the failed job [xpu / linux-jammy-xpu-2025_0-py3.9 / test (default, 1, 4, linux.idc.xpu)](https://hud.pytorch.org/pr/pytorch/pytorch/141086#33239534178) ([gh](https://github.com/pytorch/pytorch/actions/runs/11925422585/job/33239534178)) is known issue at #140917  and is fixed in PR #140916  (It's blocked by this PR to fix the build issue)


and the [Lint / lintrunner-noclang / linux-job](https://hud.pytorch.org/pr/pytorch/pytorch/141086#33237484419) ([gh](https://github.com/pytorch/pytorch/actions/runs/11925056010/job/33237484419))
>>> Lint for test/test_nestedtensor.py:

should be non related issue because this PR did not change the file test/test_nestedtensor.py.

**Review by dvrogozh — COMMENTED** [review]
@ratnampa : I suggest to update PR description and commit message by adding links to issue (#141268) and 2 commits (34b2165bdb5 and 34e420519df) which caused this issue:
```
Fixes: #141268
Fixes: 34b2165bdb5 ("Insert aten.add into fallback_ops...")
Fixes: 34e420519df ("[Reland] dont decompose baddbmm (#141045)")
```

**Review by malfet — APPROVED** [review]

**Review by dvrogozh — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h [hunk 1]
```diff
@@ -14,9 +14,11 @@ extern "C" {
...
 AOTI_TORCH_EXPORT AOTITorchError aoti_torch_xpu__fused_moving_avg_obs_fq_helper_functional(AtenTensorHandle self, AtenTensorHandle observer_on, AtenTensorHandle fake_quant_on, AtenTensorHandle running_min, AtenTensorHandle running_max, AtenTensorHandle scale, AtenTensorHandle zero_point, double averaging_const, int64_t quant_min, int64_t quant_max, int64_t ch_axis, int32_t per_row_fake_quant, int32_t symmetric_quant, AtenTensorHandle* ret0, AtenTensorHandle* ret1, AtenTensorHandle* ret2, AtenTensorHandle* ret3, AtenTensorHandle* ret4, AtenTensorHandle* ret5);
 AOTI_TORCH_EXPORT AOTITorchError aoti_torch_xpu__trilinear(AtenTensorHandle i1, AtenTensorHandle i2, AtenTensorHandle i3, const int64_t* expand1, int64_t expand1_len_, const int64_t* expand2, int64_t expand2_len_, const int64_t* expand3, int64_t expand3_len_, const int64_t* sumdim, int64_t sumdim_len_, int64_t unroll_dim, AtenTensorHandle* ret0);
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch_xpu_add_Scalar(AtenTensorHandle self, double other, double alpha, AtenTensorHandle* ret0);
```
[the remark this answers is not in the document]
Comment (EikanWang) [reply torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h:17]: @dvrogozh `aten.add.Tensor` can be well supported by the Inductor. We do not need to fall back to Aten.

**Comment by EikanWang** [comment]
The failure, `AttributeError: module 'distutils' has no attribute '_msvccompiler'`, is irrelevant with this PR. We reproduced the failure - https://github.com/pytorch/pytorch/actions/runs/11966075707/job/33361111091#step:11:493. We have filed an issue - https://github.com/pytorch/pytorch/issues/141319 and submitted a PR https://github.com/pytorch/pytorch/pull/141318 to fix it.

Since this PR is critical to address the AOTI-XPU CI failure, we will land this PR first if there are no other failures.

File: torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (etaf) [reply torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h:17]: @dvrogozh @EikanWang ,  Comparing with `aten.add.Scalar`, the `aten.add.Tensor` is registered the torch-xpu-ops as an out-of-tree op, and it's aoti API is generated in `torch/csrc/inductor/aoti_torch/generated/extend/c_shim_xpu.h`.

**Comment by guangyey** [comment]
"Unrelated failures"
@pytorchbot merge -i

File: torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (desertfire) [reply torch/csrc/inductor/aoti_torch/generated/c_shim_xpu.h:17]: > @dvrogozh `aten.add.Tensor` can be well supported by the Inductor. We do not need to fall back to Aten.

Annoyingly, it was needed for complex number.

**Comment by atalman** [comment]
@chuanqi129 please fix lint. The rest of the errors seems to be not related

**Comment by EikanWang** [comment]
@atalman , the lint error is not related with this PR as this PR does not touch the rnn.py.

**Comment by etaf** [comment]
> @chuanqi129 please fix lint. The rest of the errors seems to be not related

Hi, @atalman , The lint error happens on main branch, seems not related to this PR:
image

**Comment by guangyey** [comment]
We will rebase this PR to the latest `viable/strict` branch and land this PR once there are no other failures.

**Pull request closed without merging**